lib/puppet/provider/mount/parsed.rb

```diff
@@ -264,7 +272,12 @@ def self.mountinstances
     end
     mount_output.each do |line|
       if match = regex.match(line) and name = match.captures.first
-        instances << {:name => name, :mounted => :yes} # Only :name is important here
+        if Facter.value(:kernel) == 'Linux'
```

Comment:
Are we planning on only supporting Linux for remounting on mismatched options, or Solaris and others?